app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingDocumentsController.scala

```diff
@@ -0,0 +1,137 @@
+  val errorHandler: ErrorHandler,
+  cc: MessagesControllerComponents,
+  val config: Configuration,
+  hasSupportingDocsToUploadPage: pages.has_supporting_docs_to_upload
+)(implicit viewConfig: ViewConfig, ec: ExecutionContext)
+    extends FrontendController(cc)
+    with WithAuthAndSessionDataAction
+    with Logging
+    with SessionUpdates {
+
+  private def withSupportingDocumentChoices(
+    request: RequestWithSessionData[_]
+  )(
+    f: (
+      SessionData,
+      FillingOutReturn,
+      SingleDisposalDraftReturn,
+      UploadSupportingDocuments
+    ) => Future[Result]
+  ): Future[Result] =
+    request.sessionData.flatMap(s => s.journeyStatus.map(s -> _)) match {
+      case Some((s, r @ FillingOutReturn(_, _, _, d: SingleDisposalDraftReturn))) =>
+        d.uploadSupportingDocuments.fold[Future[Result]](
+          f(s, r, d, IncompleteUploadSupportingDocuments.empty)
+        )(f(s, r, d, _))
+      case _ => Future.successful(Redirect(controllers.routes.StartController.start()))
+    }
+
+  private def withSupportingDocsAnswers(
+    answers: UploadSupportingDocuments
+  )(f: Boolean => Future[Result]): Future[Result] =
+    answers
+      .fold(_.hasSupportingDocuments, c => Some(c.hasSupportingDocuments))
+      .fold[Future[Result]](Future.successful(Redirect(routes.UploadSupportingDocumentsController.checkYourAnswers())))(
+        f
+      )
+
+  private def commonDisplayBehaviour[A, P: Writeable, R](
+    currentAnswers: UploadSupportingDocuments
+  )(form: UploadSupportingDocuments => Form[A])(
+    page: (Form[A], Call) => P
+  )(
+    requiredPreviousAnswer: UploadSupportingDocuments => Option[R],
+    redirectToIfNoRequiredPreviousAnswer: Call
+  ): Future[Result] =
+    if (requiredPreviousAnswer(currentAnswers).isDefined) {
+      val backLink = currentAnswers.fold(
+        _ => redirectToIfNoRequiredPreviousAnswer,
+        _ => routes.UploadSupportingDocumentsController.checkYourAnswers()
+      )
+      Future.successful(Ok(page(form(currentAnswers), backLink)))
+    } else {
+      Future.successful(Redirect(redirectToIfNoRequiredPreviousAnswer))
+    }
+
+  def hasSupportingDocsToUpload() = authenticatedActionWithSessionData.async { implicit request =>
+    withSupportingDocumentChoices(request) {
+      case (_, _, _, answers) =>
+        withSupportingDocsAnswers(answers) { e =>
+          commonDisplayBehaviour(answers)(
+            form = _.fold(
+              _.hasSupportingDocuments.fold(hasSupportingDocsToUploadForm)(hasSupportingDocsToUploadForm.fill),
+              c => hasSupportingDocsToUploadForm.fill(c.hasSupportingDocuments)
+            )
+          )(
+            page = hasSupportingDocsToUploadPage(_, _)
+          )(
+            requiredPreviousAnswer = { a =>
+              a.fold(_.hasSupportingDocuments, c => Some(c.hasSupportingDocuments))
+            },
+            redirectToIfNoRequiredPreviousAnswer = routes.UploadSupportingDocumentsController.checkYourAnswers()
```

Comment:
this also defines the backlink - this should be the task list for this page. We should never get into the case where there is no required previous answer since there isn't a page previous to this in this section

Reply:
fixed